src/core/source_buffers/queued_source_buffer.ts

```diff
@@ -436,6 +464,31 @@ export default class QueuedSourceBuffer<T> {
     return this._segmentInventory.getInventory();
   }
 
+  /**
+   * Returns the list of every operations that the `QueuedSourceBuffer` is still
+   * processing. From the one with the highest priority (like the one being
+   * processed)
+   * @returns {Array.<Object>}
+   */
+  public getPendingOperations() : Array<IQSBOperation<T>> {
+    const parseQueuedOperation =
+      (op : IQSBQueueItem<T> | IQSBPendingTask<T>) : IQSBOperation<T> => {
+        // Had to be written that way for TypeScrypt
```

Comment:
wo too bad Typescript doesn't get it.

Reply:
And why not doing something like this :
```
  public getPendingOperations() : Array<IQSBOperation<T>> {
    return (this._pendingTask === null ?
      this._queue :
      ([this._pendingTask] as Array<IQSBQueueItem<T> | IQSBPendingTask<T>>)
        .concat(this._queue))
        .filter((op) => {
          return op.type === SourceBufferOperation.Push ||
                 op.type === SourceBufferOperation.Remove ||
                 op.type === SourceBufferOperation.EndOfSegment;
        });
  }
```

Reply:
(it works with Typescript)

Reply:
The basic idea was that I wanted to remove `op.subject` (as well as a `IPushTask` supplementary information) from the original value, so it was cleaner in a way to reconstruct the object.